lib/api/core/models/repositories/userRepository.js

```diff
@@ -26,11 +26,53 @@ const
   uuid = require('uuid/v4'),
   User = require('../security/user'),
   Repository = require('./repository'),
+  Request = require('kuzzle-common-objects').Request,
   {
     BadRequestError,
-    NotFoundError
+    NotFoundError,
+    InternalError: KuzzleInternalError
   } = require('kuzzle-common-objects').errors;
 
+/**
+ * @param {Kuzzle} kuzzle
+ * @param {User} user
+ */
+const removeUserStrategies = Bluebird.coroutine(function* removeUserGenerator (kuzzle, user) {
+  const
+    availableStrategies = kuzzle.pluginsManager.listStrategies(),
+    userStrategies = [],
+    errors = [],
+    request = new Request({ _id: user._id });
+
+  for (const strategy of availableStrategies) {
+    const
+      existsMethod = kuzzle.pluginsManager.getStrategyMethod(strategy, 'exists'),
+      existStrategy = yield existsMethod(request, user._id, strategy);
+
+    if (existStrategy) {
+      userStrategies.push(strategy);
+    }
+  }
+
+  if (userStrategies.length > 0) {
+    for (const strategy of userStrategies) {
+      const
+        deleteMethod = kuzzle.pluginsManager.getStrategyMethod(strategy, 'delete');
+
+      // We catch any error produced by delete as we want to make as much cleanup as possible
+      yield deleteMethod(request, user._id, strategy)
+        .catch(error => {
+          errors.push(error);
+          return Bluebird.reject(error);
+        });
+    }
+  }
+
+  if (errors.length > 0) {
+    throw new KuzzleInternalError(`was not able to remove bad credentials of user "${user._id}": ` + errors.map(error => error.message).join(';'));
```

Comment:
This should not be an internal error, since plugins are responsible of deleting credentials.
Since this is a multi-part action, with potentially multiple errors, then I would use a `PartialError`, which conveniently takes an array of errors: https://docs.kuzzle.io/plugins-reference/plugins-context/errors#partialerror
I also think that this is the right choice because that function is meant to be called by a global user removal function, which has at that point correctly deleted the user's kuid from the database.

Also, there is a weird mix of string litteral + concatenation going on here